packages/jsii/lib/assembler.ts

```diff
@@ -1624,35 +1557,20 @@ function* intersect<T>(xs: Set<T>, ys: Set<T>) {
   }
 }
 
-/**
- * Return the most constrained version given two versions
- *
- * Returns the highest version. Return undefined if the values are not pairwise
- * comparable.
- */
-function mostConstrainedVersion(version1: string, version2: string): string | undefined {
-  if (semver.satisfies(version1, `^${version2}`)) {
-    // If v1 satisifies v2, then either:
-    // - v2 also satisfies v1, in which case it doesn't matter which we return
-    // - v2 does not satisfy v1, in which it must be the case that v1 is higher
-    return version1;
-  }
-
-  // Reverse logic
-  if (semver.satisfies(version2, `^${version1}`)) { return version2; }
-
-  return undefined;
-}
-
-function flatten<T>(xs: T[][]): T[] {
-  return Array.prototype.concat.call([], ...xs);
-}
-
 function noEmptyDict<T>(xs: {[key: string]: T}): {[key: string]: T} | undefined {
   if (Object.keys(xs).length === 0) { return undefined; }
   return xs;
 }
 
+function toDependencyClosure(assemblies: readonly spec.Assembly[]): { [name: string]: spec.AssemblyConfiguration } {
+  const result: { [name: string]: spec.AssemblyTargets } = {};
```

Comment:
I see you got rid of the version conflict checkage. Are you sure it can't happen?

Reply:
That's a very good point. I somehow though it was no longer necessary but that's not true. I'll update with a modified version where the "no conflict" check happens upfront (when loading the `ProjectInfo` object) and reduces the dependency statement to the intersection of all requirements found in the closure (which is the "most correct" way to go here, I feel, and will detect non-overlapping requirements at the same time)